src/jit/codegenxarch.cpp

```diff
@@ -4612,7 +4612,62 @@ void CodeGen::genCodeForStoreLclVar(GenTreeLclVar* tree)
             else if (op1->gtRegNum != targetReg)
             {
                 assert(op1->gtRegNum != REG_NA);
-                emit->emitInsBinary(ins_Move_Extend(targetType, true), emitTypeSize(tree), tree, op1);
+
+                instruction ins     = ins_Move_Extend(targetType, true);
+                bool        canSkip = false;
+
+                // Check for skippable if we're optimizing, we're within an IG, and we have a simple mov.
+                //
+                if (compiler->opts.OptimizationEnabled() && (emit->emitCurIGinsCnt > 0) && (ins == INS_mov))
+                {
+                    emitter::instrDesc* prevInstr = emit->emitLastIns;
+                    assert(prevInstr != nullptr);
+
+                    // See if we're about to emit the same or similar kind of mov.
+                    //
+                    // Sure would be nice to just provisionally emit the new instruction and compare
+                    // instrDescs, but that's not possible just yet.
+                    //
+                    // Do we need any other safety checks here? Maybe same gcness?
```

Comment:
It's been a while since I dealt with the gc reporting, but I *think* it's the case that the emitter does the final reporting, so I don't think there'd be an impact because we haven't actually emitted this instruction yet, and the gcness of the registers isn't changing.